### PR TITLE
Add option to change device type

### PR DESCRIPTION
**Pull request by rahul-thakoor**
Change-type: minor

### CODE REVIEW

File: src/stage1/migrate_info.rs [hunk 1]
```diff
@@ -241,6 +256,7 @@ impl MigrateInfo {
             nwmgr_files,
             system_proxy_files,
             backup,
+            change_dt_to: opts.change_dt_to().to_owned().unwrap_or("".to_string()),
```
Comment (lmbarros) [inline src/stage1/migrate_info.rs:259]: If I followed it right, this value ends up in the stage 2 config. So, when dealing with this, stage 2 will need to check if the value is an empty string or a valid DT. I think using an `Option` would be clearer ("if the value is provided, it is a DT for sure") and more idiomatic.

Comment (rahul-thakoor) [reply src/stage1/migrate_info.rs]: Yes, still WIP :)

I realised i don't need it in MigrationInfo, but only need to carry the information into Stage2, so I simplified it a bit

**Review by lmbarros — COMMENTED** [review]
Overall looks good, but I left a couple smallish suggestions.

File: src/stage1/api_calls.rs [hunk 2]
```diff
@@ -132,3 +146,99 @@ pub(crate) fn get_os_image(
...
+    let dt_id_request_url =  format!("{api_endpoint}{DEVICE__TYPE_URL_ENDPOINT}?$orderby=name%20asc&$top=1&$select=id&$filter=device_type_alias/any(dta:dta/is_referenced_by__alias%20eq%20%27{dt_slug}%27)");
+
+    debug!(
+        "patch_device_type: dt_id_request_url: '{}'",
+        dt_id_request_url
+    );
+
+    let res = Client::builder()
+        .default_headers(headers.clone())
+        .build()
```
[the remark this answers is not in the document]
Comment (rahul-thakoor) [reply src/stage1/api_calls.rs:173]: yeah we should look into this separately

**Review by lmbarros — APPROVED** [review]
:rocket:

**Pull request merged**